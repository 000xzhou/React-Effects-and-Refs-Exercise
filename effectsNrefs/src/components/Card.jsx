function Card({ image, name }) {
  return (
    <div>
      <img src={image} alt={name} />
    </div>
  );
}

export default Card;
