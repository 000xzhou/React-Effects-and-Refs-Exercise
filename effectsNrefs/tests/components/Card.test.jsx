import { describe, it, expect } from "vitest";
import { render, fireEvent } from "@testing-library/react";
import Card from "../../src/components/Card";

describe("Card", () => {
  it("renders the Card component", () => {
    render(<Card />);
  });

  it("should match snapshot", () => {
    const { asFragment } = render(
      <Card
        image="https://www.bluecross.org.uk/sites/default/files/assets/images/124044lpr.jpg"
        name="cat"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
