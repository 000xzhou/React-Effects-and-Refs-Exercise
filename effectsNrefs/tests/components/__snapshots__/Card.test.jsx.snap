// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Card > should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <img
      alt="cat"
      src="https://www.bluecross.org.uk/sites/default/files/assets/images/124044lpr.jpg"
    />
  </div>
</DocumentFragment>
`;
